aten/src/ATen/native/TensorShape.cpp

```diff
@@ -128,6 +138,14 @@ Tensor & _cat_out_cpu(Tensor& result, TensorList tensors, int64_t dim) {
   result_size[dim] = cat_dim_size;
   result.resize_(result_size);
 
+  // fast path for single thread when inputs are contiguous and not empty
+  bool use_serial_kernel = result.numel() < at::internal::GRAIN_SIZE || at::get_num_threads() == 1;
+  ScalarType dtype = notSkippedTensor.scalar_type();
+  if (use_serial_kernel && allContiguous && (dtype == ScalarType::Double || dtype == ScalarType::Float)) {
```

Comment:
result should be contiguous, or you will have incorrect data here.

```python
x = torch.randn((4, 3, 8, 8 ))                                                                                                                                                                                                        
y = torch.randn(x.shape)                                                                                                                                                                                                              
z = torch.cat((x,y))                                                                                                                                                                                                                  
w = z.clone().contiguous(memory_format=torch.channels_last)                                                                                                                                                                           
w.stride()  # (192, 1, 24, 3)                                                                                                                                                                                                                      
u = torch.cat((x,y),out=w)
(abs(w-z)).max() # >0!
```

Reply:
OK, i see. Do we need add new test cases to cover this?

Reply:
Yes please.

Reply:
Fixed. Test case added.